Etabs_Adapter/AdapterActions/Exectute.cs

```diff
@@ -0,0 +1,174 @@
+
+        public bool RunCommand(Save command)
+        {
+            return m_model.File.Save() == 0;
+        }
+
+        /***************************************************/
+
+        public bool RunCommand(SaveAs command)
+        {
+            return m_model.File.Save(command.FileName) == 0;
+        }
+
+        /***************************************************/
+
+        public bool RunCommand(Open command)
+        {
+            bool success = m_model.File.OpenFile(command.FileName) == 0;
+            success &= m_model.SetPresentUnits(eUnits.N_m_C) == 0;
+            return success;
+        }
+
+        /***************************************************/
+
+        public bool RunCommand(Analyse command)
+        {
+            bool success = m_model.Analyze.SetRunCaseFlag("", true, true) == 0;
+
+            if (!success)
+            {
+                Engine.Reflection.Compute.RecordWarning("Failed to set up cases to run. Model has not been analysed");
+                return false;
+            }
+
+            success &= m_model.Analyze.RunAnalysis() == 0;
+            return success;
+        }
+
+        /***************************************************/
+
+        public bool RunCommand(AnalyseLoadCases command)
+        {
+            bool success;
+            var cases = command.LoadCases;
+
+            if (cases == null)
+            {
+                Engine.Reflection.Compute.RecordNote("No cases provided, all cases will be run");
+                success = m_model.Analyze.SetRunCaseFlag("", true, true) == 0;
+            }
+            else
+            {
+                //Unselect all cases
+                success = m_model.Analyze.SetRunCaseFlag("", false, true) == 0;
+                
+                //Select provided cases
+                foreach (object item in cases)
+                {
+                    string name;
+                    if (item == null)
+                        continue;
+                    if (item is string)
+                        name = item as string;
+                    else if (item is ICase)
+                        name = (item as ICase).Name;
+                    else
+                    {
+                        Engine.Reflection.Compute.RecordWarning("Can not set up cases for running of type " + item.GetType().Name + ". Item " + item.ToString() + " will be ignored. Please provide case names or BHoM cases to be run");
+                        continue;
+                    }
+
+                    bool caseSuccess = m_model.Analyze.SetRunCaseFlag(name, true, false) == 0;
+                    success &= caseSuccess;
+
+                    if (!caseSuccess)
+                    {
+                        Engine.Reflection.Compute.RecordWarning("Failed to set case " + name + "for running. Please check that the case exists in the model");
+                    }
+                }
+
+            }
+
+            success &= m_model.Analyze.RunAnalysis() == 0;
```

Comment:
If the model isn't saved this returns `false` (and doesn't run). Would prefer if there was a check beforehand with appropriate error message. Example:  Testing what `cSapModel.GetModelFilename()` returns (haven't tested)

Reply:
tried your suggestion and think it could work. Update pushed

Reply:
Works, but as a fun fact, ETABS lets you save as `(Untitled).EDB` but doesn't believe you afterwards